book/src/state_specific_systems/how_to_define_state_dispatcher_external.md

```diff
@@ -1,156 +1,277 @@
+# /// This allows you to specify which `System`s you want to run within the state.
+# pub struct CustomDispatcherStateBuilder<'a, 'b> {
+#     /// The `State` specific `DispatcherBuilder`.
+#     dispatcher_builder: DispatcherBuilder<'a, 'b>,
+# }
+# 
+# impl<'a, 'b> CustomDispatcherStateBuilder<'a, 'b> {
+#     /// Returns a new `CustomDispatcherStateBuilder`.
+#     pub fn new() -> Self {
+#         CustomDispatcherStateBuilder {
+#             dispatcher_builder: DispatcherBuilder::new(),
+#         }
+#     }
+# 
+#     /// Registers a `System` with the dispatcher builder.
+#     ///
+#     /// # Parameters
+#     ///
+#     /// * `system`: `System` to register.
+#     /// * `name`: Name to register the system with, used for dependency ordering.
+#     /// * `deps`: Names of systems that must run before this system.
+#     pub fn with<Sys>(mut self, system: Sys, name: &str, deps: &[&str]) -> Self
+#         where Sys: for<'c> System<'c> + Send + 'a,
+#     {
+#         self.dispatcher_builder.add(system, name, deps);
+#         self
+#     }
+# 
+#     /// Builds and returns the `CustomDispatcherState`.
+#     pub fn build(self) -> CustomDispatcherState<'a, 'b> {
+#         CustomDispatcherState::new(self.dispatcher_builder)
+#     }
+# }
+# 
+# pub struct CustomDispatcherState<'a, 'b> {
+#     /// The `State` specific `DispatcherBuilder`. This will be used to build the actual dispatcher.
+#     dispatcher_builder: Option<DispatcherBuilder<'a, 'b>>,
+# 
+#     /// The `State` specific `Dispatcher`, containing `System`s only relevant for this `State`.
+#     dispatcher: Option<Dispatcher<'a, 'b>>,
+# }
+# 
+impl<'a, 'b> SimpleState for CustomDispatcherState<'a, 'b> {
+    fn on_start(&mut self, data: StateData<GameData>) {
+        if self.dispatcher.is_none() {
+            let mut dispatcher = self.dispatcher_builder
+                .take()
+                .expect("Expected `dispatcher_builder` to exist when `dispatcher` is not yet built.")
+                .build();
+            dispatcher.setup(&mut data.world.res);
+            self.dispatcher = Some(dispatcher);
+        }
+    }
+
+    fn update(&mut self, data: &mut StateData<GameData>) -> SimpleTrans {
+        self.dispatcher.as_mut().unwrap().dispatch(&data.world.res);
+
+        Trans::None
+    }
+}
+# 
+# impl<'a, 'b> CustomDispatcherState<'a, 'b> {
+#     fn new(dispatcher_builder: DispatcherBuilder<'a, 'b>) -> Self {
+#         Self {
+#             dispatcher_builder: Some(dispatcher_builder),
+#             dispatcher: None,
+#         }
+#     }
+# }
+```
+
+Now, any systems in the state-specific `Dispatcher` will only run in this `State`. The *application* `Dispatcher` is automatically executed for `SimpleState` implementations. For more complex states that implement other `State` specific traits, the *application* `Dispatcher` has to be explicitly executed on top of the `State` specific `Dispatcher`. This is done by slightly modifying your `State`s `update(..)` method:
```

Comment:
I think you should just get rid of this part. People should know if they use a custom state that they need to update the main dispatcher as well.

Reply:
Ahh, that's still some old documentation that I haven't fixed up yet. Ignore that one. :)

Reply:
I'm not sure how I feel about that. I prefer too much over too little and I ran into an issue when I followed the old guide that was based on custom states. I'd personally keep it in just to be sure but I'll let you guys decide!

Reply:
I ended up removing that part in my current version.